tests/test_full_nlp.py

```diff
@@ -0,0 +1,198 @@
+                                 device=device)
+    finetuning_trainer.fit()
+    finetuning_trainer.eval(finetuning_eval_dataloader)
+
+    loaded_finetuning_trainer = Trainer(model=finetuning_model_copy,
+                                        load_path=str(tmp_path / 'finetuning_checkpoints' / 'latest-rank0.pt'),
+                                        seed=17,
+                                        algorithms=algorithms,
+                                        device=device)
+
+    loaded_finetuning_trainer.eval(finetuning_eval_dataloader)
+
+    original_acc = finetuning_trainer.state.eval_metrics['eval']['Accuracy']
+    loaded_acc = loaded_finetuning_trainer.state.eval_metrics['eval']['Accuracy']
+    assert original_acc.compute() > 0.0
+    assert original_acc.compute() == loaded_acc.compute()
+
+    return loaded_finetuning_trainer, finetuning_eval_dataloader
+
+
+def inference_test_helper(model, original_input, original_output, tmp_path, save_format, device):
+    os.mkdir(tmp_path / 'inference_checkpoints')
+    sample_input = (original_input, {})
+
+    inference.export_for_inference(model=model,
+                                   save_format=save_format,
+                                   save_path=str(tmp_path / 'inference_checkpoints' / f'exported_model.{save_format}'),
+                                   sample_input=sample_input)
+
+    copied_batch = copy.deepcopy(original_input)
+
+    if save_format == 'onnx':
+        onnx = pytest.importorskip('onnx')
+        ort = pytest.importorskip('onnxruntime')
+        loaded_inference_model = onnx.load(str(tmp_path / 'inference_checkpoints' / 'exported_model.onnx'))
+        onnx.checker.check_model(loaded_inference_model)
+        ort_session = ort.InferenceSession(str(tmp_path / 'inference_checkpoints' / 'exported_model.onnx'))
+
+        for key, value in copied_batch.items():
+            copied_batch[key] = value.numpy()
+        loaded_model_out = ort_session.run(None, copied_batch)
+    elif save_format == 'torchscript':
+        loaded_inference_model = torch.jit.load(str(tmp_path / 'inference_checkpoints' / 'exported_model.torchscript'))
+        loaded_inference_model.eval()
+        loaded_model_out = loaded_inference_model(copied_batch)
+    else:
+        raise ValueError('Unsupported save format')
+
+    torch.testing.assert_close(
+        loaded_model_out[1] if isinstance(loaded_model_out, list) else loaded_model_out.detach().numpy(),
+        original_output.detach().numpy()
+        if isinstance(original_output, torch.Tensor) else original_output.logits.detach().numpy())
+
+
+@device('cpu', 'gpu')
+@pytest.mark.parametrize('model_type,algorithms,save_format', [('tinybert_hf', [GatedLinearUnits()], 'onnx'),
+                                                               ('simpletransformer', [], 'torchscript')])
+def test_full_nlp_pipeline(model_type, algorithms, save_format, tiny_bert_tokenizer, tmp_path, request, device):
```

Comment:
Would add a comment at the top of this test explaining it a little more, since it is long & includes multiple helpers. Will be easier for others coming across this file later on.